colossalai/inference/core/engine.py

```diff
@@ -445,15 +450,19 @@ def generate(
             List[str]: Inference result returned by one generation.
         """
         with torch.inference_mode():
+            if generation_config is not None:
+                self.generation_config = generation_config
+
             if prompts is not None or prompts_token_ids is not None:
+                if isinstance(prompts, str) and isinstance(request_ids, int):
+                    prompts = [prompts]
+                    request_ids = [request_ids]
```

Comment:
These have been included in `add_request`. Check the usage of the function

Reply:
fixed, added logic for `request_ids`